### PR TITLE
chore: add shortcuts

### PR DESCRIPTION
**Pull request by jrandolf**

### CODE REVIEW

File: src/bidiMapper/domains/input/ActionDispatcher.ts [hunk 1]
```diff
@@ -569,6 +584,38 @@ export class ActionDispatcher {
...
+          break;
+        case 'KeyC':
+          command = 'Copy';
+          break;
+        case 'KeyV':
+          if (source.shift) {
+            command = 'PasteAndMatchStyle';
+          } else {
+            command = 'Paste';
+          }
```
Comment (thiagowfx) [inline src/bidiMapper/domains/input/ActionDispatcher.ts:604]: optional:

command = source.shift ? '...' : '...'

ditto below

File: src/bidiMapper/domains/input/ActionDispatcher.ts [hunk 2]
```diff
@@ -69,14 +73,25 @@ async function getElementCenter(
   return {x: x as number, y: y as number};
 }
 
+async function isMacOS(context: BrowsingContextImpl) {
+  const {result} = await (
+    await context.getOrCreateSandbox(undefined)
+  ).callFunction(IS_MAC_DECL, {type: 'undefined'}, [], false, 'none', {});
+  assert(result.type !== 'exception');
+  assert(result.result.type === 'boolean');
```
Comment (thiagowfx) [inline src/bidiMapper/domains/input/ActionDispatcher.ts:81]: What is the side effect of this assert during runtime? Would the mapper tab crash?

File: src/bidiMapper/domains/input/ActionDispatcher.ts [hunk 3]
```diff
@@ -39,6 +39,10 @@ const CALCULATE_IN_VIEW_CENTER_PT_DECL = ((i: Element) => {
   return [e + ((n - e) >> 1), h + ((m - h) >> 1)];
 }).toString();
 
+const IS_MAC_DECL = (() => {
+  return navigator.platform.toLowerCase().includes('mac');
+}).toString();
```
Comment (thiagowfx) [inline src/bidiMapper/domains/input/ActionDispatcher.ts:44]: I'd just inline this. If you want to preserve syntax highlighting in text editors, then do:

```
String(() => {
  return ...
})
```

We use this pattern elsewhere

File: src/bidiMapper/domains/input/ActionDispatcher.ts (hunk 2 above)
Comment (jrandolf) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:81]: It throws an unknown error.

File: src/bidiMapper/domains/input/ActionDispatcher.ts (hunk 3 above)
Comment (jrandolf) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:44]: Inlining this causes the formatter to modify the declaration's call site drastically. If allowed, I'd prefer to keep it this way consistent with the above decl.

File: src/bidiMapper/domains/input/ActionDispatcher.ts [hunk 4]
```diff
@@ -569,6 +584,38 @@ export class ActionDispatcher {
...
+          command = 'Cut';
+          break;
+        case 'KeyZ':
+          if (source.shift) {
+            command = 'Redo';
+          } else {
+            command = 'Undo';
+          }
+          break;
+        default:
```
[the remark this answers is not in the document]
Comment (jrandolf) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:616]: It won't be exhaustive then and the linter will throw.

[the remark this answers is not in the document]
Comment (thiagowfx) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:608]: Then could you add a comment below?

`// Intentionally left empty` or similar

File: src/bidiMapper/domains/input/ActionDispatcher.ts (hunk 3 above)
Comment (thiagowfx) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:44]: OK

File: src/bidiMapper/domains/input/ActionDispatcher.ts [hunk 5]
```diff
@@ -69,14 +73,25 @@ async function getElementCenter(
...
 export class ActionDispatcher {
   #tickStart = 0;
   #tickDuration = 0;
   #inputState: InputState;
   #context: BrowsingContextImpl;
+  #isMacOS: Promise<boolean>;
   constructor(inputState: InputState, context: BrowsingContextImpl) {
     this.#inputState = inputState;
     this.#context = context;
+    this.#isMacOS = isMacOS(this.#context);
```
Comment (OrKoN) [inline src/bidiMapper/domains/input/ActionDispatcher.ts:94]: here an unhandled promise rejection might occur: please add a catch

Comment (jrandolf) [reply src/bidiMapper/domains/input/ActionDispatcher.ts:94]: In what situation?

Comment (OrKoN) [reply src/bidiMapper/domains/input/ActionDispatcher.ts]: when the async calls inside isMacOS throw (for example, target shutting down or any other reason really)

**Pull request merged**